mobile/src/main/java/butter/droid/ui/preferences/PreferencesActivity.java

```diff
@@ -218,6 +243,18 @@ public void openChangelog() {
         startActivity(AboutActivity.getIntent(this));
     }
 
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+
+        if (requestCode == OPEN_DIRECTORY_SELECTOR_PERMISSION){
+            if (grantResults[0] == PackageManager.PERMISSION_GRANTED){
+                openDirectorySelector(Prefs.STORAGE_LOCATION, R.string.storage_location, preferencesHandler.getStorageLocation());
```

Comment:
this is an horrible solution. here I had to copy-paste the parameters from the `BasePreferencesPresenter`